wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -414,8 +413,7 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
       }
 
       // If pre-condition is set, then evaluate the precondition
-      if (PRECONDITION_LANGUAGE_JEXL.equalsIgnoreCase(config.getPreconditionLanguage())
-          && checkPreconditionNotEmpty(false)) {
+      if (!checkSQLExecution(config) && checkPreconditionNotEmpty(false)) {
```

Comment:
Only concern here, we are checking SQL execution for every row. This is not efficient. Can we have a global variable and initialize it in `initialize` function. afaik `initialize` is executed on all spark nodes.